tools/toolchains/arm.py

```diff
@@ -81,6 +91,9 @@ def __init__(self, target, notify=None, macros=None,
             if "--library_type=microlib" not in self.flags['common']:
                 self.flags['common'].append("--library_type=microlib")
 
+        if getattr(target, "default_lib", "std") == "small":
```

Comment:
Perhaps you should do this:
```python

        if target.default_lib == "small":
            if "--predefine=-D__MICROLIB" not in self.flags['ld']:
                self.flags['ld'].append("--predefine=-D__MICROLIB")
        
```

Reply:
As explained in the previous comment we need getattr

Reply:
https://github.com/rajkan01/mbed-os/pull/4#discussion_r346270258